src/sentry/integrations/source_code_management/repository.py

```diff
@@ -0,0 +1,200 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import Collection, Mapping, Sequence
+from typing import Any
+
+import sentry_sdk
+
+from sentry.auth.exceptions import IdentityNotValid
+from sentry.integrations.services.integration import integration_service
+from sentry.integrations.services.integration.model import RpcIntegration
+from sentry.integrations.services.repository import RpcRepository, repository_service
+from sentry.models.identity import Identity
+from sentry.models.integrations.integration import Integration
+from sentry.models.repository import Repository
+from sentry.shared_integrations.client.base import BaseApiResponseX
+from sentry.shared_integrations.exceptions import ApiError, IntegrationError
+
+
+class RepositoryIntegration(ABC):
+    repo_search = False
+    model: RpcIntegration | Integration
+
+    # @abstractmethod
+    @property
+    def codeowners_locations(self) -> Sequence[str] | None:
```

Comment:
could probably make this abstract and return None everywhere that doesn't support codeowners